File: client/src/utils/formatters.ts
export const formatPrice = (price: number): string => {
  return new Intl.NumberFormat('en-IN', {
    style: 'currency',
    currency: 'INR',
    maximumFractionDigits: 0,
  }).format(price);
};

export const formatCurrency = formatPrice;

export const formatTimeAgo = (dateString: string): string => {
  const date = new Date(dateString);
  const now = new Date();
  const diffInSeconds = Math.floor((now.getTime() - date.getTime()) / 1000);

  if (diffInSeconds < 60) return 'Just now';
  if (diffInSeconds < 3600) return `${Math.floor(diffInSeconds / 60)}m ago`;
  if (diffInSeconds < 86400) return `${Math.floor(diffInSeconds / 3600)}h ago`;
  if (diffInSeconds < 604800) return `${Math.floor(diffInSeconds / 86400)}d ago`;
  return date.toLocaleDateString('en-IN', { month: 'short', day: 'numeric' });
};

export const getCategoryBadge = (category: string): string => {
  const map: Record<string, string> = {
    DRAWING_TOOLS: '📐 Drawing Tools',
    ELECTRONICS: '💻 Electronics',
    BOOKS_NOTES: '📚 Books & Notes',
    FASHION: '👕 Fashion',
    HOSTEL_REQ: '🛏️ Hostel & Req',
    HOBBY_SPORT: '🏸 Hobby / Sport',
    OTHERS: '📦 Others',
    // Legacy mappings
    LAB_STATIONERY: '📐 Drawing Tools',
    BOOKS_ACADEMICS: '📚 Books & Notes',
    HOSTEL_ESSENTIALS: '🛏️ Hostel & Req',
    CYCLES: '🚲 Cycles',
    SPORTS_FITNESS: '🏸 Hobby / Sport',
    OTHER: '📦 Others',
  };
  return map[category] || category;
};

File: mobile/lib/core/utils/formatters.dart
import 'package:intl/intl.dart';

class Formatters {
  static String formatPrice(num price) {
    final formatter = NumberFormat.currency(
      locale: 'en_IN',
      symbol: '₹',
      decimalDigits: 0,
    );
    return formatter.format(price);
  }

  static String formatTimeAgo(dynamic dateInput) {
    if (dateInput == null) return 'recently';
    DateTime date;
    if (dateInput is DateTime) {
      date = dateInput;
    } else {
      date = DateTime.tryParse(dateInput.toString()) ?? DateTime.now();
    }

    final diff = DateTime.now().difference(date);
    if (diff.inDays > 30) {
      return DateFormat('d MMM').format(date);
    } else if (diff.inDays >= 1) {
      return '${diff.inDays}d ago';
    } else if (diff.inHours >= 1) {
      return '${diff.inHours}h ago';
    } else if (diff.inMinutes >= 1) {
      return '${diff.inMinutes}m ago';
    }
    return 'Just now';
  }

  static String getCategoryName(String categoryId) {
    switch (categoryId) {
      case 'DRAWING_TOOLS':
        return 'Drawing Tools';
      case 'ELECTRONICS':
        return 'Electronics';
      case 'BOOKS_NOTES':
        return 'Books & Notes';
      case 'HOSTEL_REQ':
        return 'Hostel & Req';
      case 'FASHION':
        return 'Fashion';
      case 'HOBBY_SPORT':
        return 'Hobby / Sport';
      default:
        return 'Others';
    }
  }

  static String getConditionLabel(String condition) {
    switch (condition) {
      case 'NEW':
        return 'Brand New';
      case 'LIKE_NEW':
        return 'Like New';
      case 'GOOD':
        return 'Good';
      case 'FAIR':
        return 'Fair';
      default:
        return condition;
    }
  }
}
